src/cript/nodes/primary_nodes/material.py

```diff
@@ -445,27 +445,24 @@ def property(self, new_property_list: List[Any]) -> None:
 
     @classmethod
     def _from_json(cls, json_dict: dict):
-        # TODO find a better way to know identifiers
-        all_identifier_list = [
-            "amino_acid",
-            "bigsmiles",
-            "cas",
-            "chem_formula",
-            "chem_repeat",
-            "chemical_id",
-            "inchi",
-            "inchi_key",
-            "lot_number",
-            "mol_form",
-            "names",
-            "pubchem_cid",
-            "smiles",
-            "vendor",
-        ]
+
+        from cript.api.api import _get_global_cached_api
+        api = _get_global_cached_api()
+
+        # get JSON vocabulary of all "material_identifier_key"
+        all_identifiers: List[dict] = api.get_vocab_by_category("material_identifier_key")
+
+        # placeholder for what eventually will be all the names of material identifiers
+        # e.g. "smiles", "bigsmiles", etc.
+        all_identifiers_list: List[str] = []
+
+        # loop through and take all "name" from every dict and append it to `all_identifiers_list`
+        for identifier in all_identifiers:
+            all_identifiers_list.append(identifier.get("name"))
 
         identifier_argument = []
-        # Convert identifiers to to a list
-        for identifier in all_identifier_list:
+        # Convert identifiers to a list
+        for identifier in all_identifiers_list:
```

Comment:
Yes, this is great.
Automatically assembling the possible list.
We rely on the vocab to be complete here, but I think as is a fair assumption.
It would be a serious bug of the back end if the material allows an identifier, that is not in the vocab.

Reply:
yeah I think that is mostly okay, if something is wrong with the backend that is out of our hands